### PR TITLE
De-duplicate code in user and home page

### PR DESCRIPTION
**Pull request by baqirkhan**
Solves this: https://github.com/denoland/saaskit/issues/232

Also makes some minor improvement where some unsed imports  and unnecceasry `await`'s are removed.
- `deno task ok` runs successfully.

### CODE REVIEW

File: routes/user/[username].tsx [hunk 1]
```diff
@@ -9,31 +9,19 @@ import ItemSummary from "@/components/ItemSummary.tsx";
 import {
   getItemsByUserId,
   getUserByLogin,
-  getUserBySessionId,
-  getVotedItemIdsByUser,
   type Item,
   type User,
 } from "@/utils/db.ts";
 import { pluralize } from "../../utils/display.ts";
```
Comment (iuioiua) [inline routes/user/[username].tsx:15]:
```suggestion
import { pluralize } from "@/utils/display.ts";
```

File: routes/index.tsx [hunk 2]
```diff
@@ -7,13 +7,13 @@ import type { State } from "./_middleware.ts";
...
   getAllItems,
-  getUserBySessionId,
   getUsersByIds,
-  getVotedItemIdsByUser,
   incrementVisitsPerDay,
   type Item,
   type User,
 } from "@/utils/db.ts";
+import compareScore from "../utils/compareScore.ts";
+import { getVotedItemsBySessionUser } from "../utils/db.ts";
```
Comment (iuioiua) [inline routes/index.tsx:16]:
```suggestion
import { getVotedItemsBySessionUser } from "@/utils/db.ts";
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Looks mostly good! Just a few tweaks are needed.

File: utils/compareScore.ts
[the remark this answers is not in the document]
Comment (baqirkhan) [reply utils/compareScore.ts:1]: Makes sense!

File: .gitignore
[the remark this answers is not in the document]
Comment (baqirkhan) [reply .gitignore:1]: Reverted.
Could you please explain on why we can't have `.idea/` or `.vscode` files here? `git diff`on my IDE wants to add these files.
image

[the remark this answers is not in the document]
Comment (iuioiua) [reply .gitignore:1]: Playing with ideas in a `.idea` folder isn't a common enough practice.

File: utils/db.ts [hunk 3]
```diff
@@ -406,6 +406,22 @@ export async function getVisitsPerDay(date: Date) {
   return res.value;
 }
 
+export async function getVotedItemsBySessionUser(
+  sessionId: string | undefined,
```
Comment (iuioiua) [inline utils/db.ts:410]:
```suggestion
  sessionId?: string,
```

File: utils/db.ts [hunk 4]
```diff
@@ -406,6 +406,22 @@ export async function getVisitsPerDay(date: Date) {
...
+) {
+  let votedItemIds: string[] = [];
+
+  if (sessionId) {
+    const sessionUser = await getUserBySessionId(sessionId!);
+    if (sessionUser) {
+      votedItemIds = await getVotedItemIdsByUser(sessionUser!.id);
+    }
+  }
+  /** @todo Optimise */
```
Comment (iuioiua) [inline utils/db.ts:421]:
```suggestion
```

File: utils/db.ts [hunk 5]
```diff
@@ -406,6 +406,22 @@ export async function getVisitsPerDay(date: Date) {
   return res.value;
 }
```
Comment (iuioiua) [inline utils/db.ts:408]:
```suggestion
/** @todo Optimise */
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
I made a couple of tweaks, and now it LGTM! Thank you, @baqirkhan.

**Pull request merged**